### ISSUE TITLE
Convert integer date value to an actual `Date`

### ISSUE THREAD

**Issue by elvanja** [xavier/xlsx_reader#20]
In the Excel sheet I have a date column for which I only get integer values, e.g. 41760. If I understand Excel correctly I can just do `Date.new!(1900, 1, 1) |> Date.add(41760)` and get `~D[2014-05-03]` as the actual date. Is that correct? Maybe there is a nicer way to convert those integers into dates?

**Comment by xavier** [comment]
If the Excel sheet uses some [standard cell format](https://github.com/xavier/xlsx_reader/blob/master/lib/xlsx_reader/styles.ex#L42-L48) for the date, the library should pick that up and transform the serial date into a `Date` struct automatically.

We also try to detect some [common date/time formats](https://github.com/xavier/xlsx_reader/blob/master/lib/xlsx_reader/styles.ex#L79-L88). Perhaps that your spreadsheet is not using any of those. In that case, you [may specify your own cell format](https://github.com/xavier/xlsx_reader/blob/master/lib/xlsx_reader.ex#L57-L94).

For various reason, the conversion is not that trivial so you may want to use [`XlsxReader.Conversion.to_date/1`](https://github.com/xavier/xlsx_reader/blob/master/lib/xlsx_reader/conversion.ex#L240).

I hope this helps.

**Comment by elvanja** [comment]
Yeah, I saw the custom formats / hints section but didn't figure out how to use that. E.g. if I read it all correctly, one can supply a set of options, see https://github.com/xavier/xlsx_reader/blob/master/lib/xlsx_reader/parsers/worksheet_parser.ex#L36-L54, but none resemble those hints from the docs. I am likely missing something.

Anyway, I tried also with `type_conversion: false`, like this:
```elixir
data_file = ".../some_file.xlsx"
{:ok, package} = XlsxReader.open(data_file)
{:ok, [columns | rows]} = XlsxReader.sheet(package, "Data Import", type_conversion: false)
```

And when I read the date field, assuming that `type_conversion: false` returns raw data (no conversions), I get value `"41760"` (string with integer value 41760). But then this is interesting:

```elixir
iex> XlsxReader.Conversion.to_date(start_date)
{:ok, ~D[2014-05-01]}

iex> Date.new!(1900, 1, 1) |> Date.add(String.to_integer("41760"))
~D[2014-05-03]
```

After reading http://www.cpearson.com/excel/datetime.htm, which you nicely provided in code, it became clear and indeed [XlsxReader.Conversion.to_date/1](https://github.com/xavier/xlsx_reader/blob/master/lib/xlsx_reader/conversion.ex#L240) serves my purpose well.

So, I can get around the issue, thanks for the tips! Would be nice to know how to use those conversion hints though. Thank you for your time! 🙇🏼

**Comment by xavier** [comment]
Glad to hear that you solved your problem.

If you want to use custom cell formats, you must pass them when you load the package:

```elixir
XlsxReader.open("test.xlsx", supported_custom_formats: [{"yyyy, mm dd", :date}])
```
